### PR TITLE
[STRAT-72] Dynamics refactor (cleaned rolldown data)

### PR DESCRIPTION
**Pull request by ufshk**
Rolldown data has been cleaned. Exists in both original (raw) files and cleaned files in dynamics/rolldowndata/rolldown/processed.

Cleaned CSV has columns: Time Offset [ms], vehicle_velocity_left, vehicle_velocity_right, average_velocity

### CODE REVIEW

File: dynamics/rolldowndata/rolldown/cleaned_RollDown2019-06-02-1.csv [hunk 1]
```diff
@@ -0,0 +1,219 @@
+Time Offset [ms],voltage,current,vehicle_velocity_left,vehicle_velocity_right,average_velocity
+9257.388,,,,,
```
Comment (clarkevandenhoven) [inline dynamics/rolldowndata/rolldown/cleaned_RollDown2019-06-02-1.csv:2]: This is really close, just get rid of all of the ones without a velocity reading :)

**Comment by ufshk** [comment]
I believe this is still not correct despite the changes I've made. Will look over the logic one more time.

**Comment by ufshk** [comment]
> I believe this is still not correct despite the changes I've made. Will look over the logic one more time.

found the issue - PR should be fully ready to go

File: dynamics/rolldowndata/rolldown/clean_rolldown_data.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    # cumsum on boolean column to find multiple
+    # subsets of rows fitting our requirements
+    find_subset_bool = find_subset['throttle and brake == 0 or na'] != find_subset['throttle and brake == 0 or na'].shift()
+    find_subset_cumsum = find_subset_bool.cumsum()
+    subset_groups = find_subset.groupby(find_subset_cumsum)
+    group_counts = subset_groups.agg({'row_no':['count', 'min', 'max']})
+
+    # drop unnecessary column levels
+    group_counts.columns = group_counts.columns.droplevel()
```
Comment (e-wai) [inline dynamics/rolldowndata/rolldown/clean_rolldown_data.py:57]: what columns does this drop?

File: dynamics/rolldowndata/rolldown/clean_rolldown_data.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    f = join('.', unprocessed_path, f)
+    file_paths.append(f)
+
+# clean and transform each file in file_paths list
+for i, f in enumerate(file_paths):
+    rolldown = pd.read_csv(f)
+
+    # make a new boolean column, needed to clean data later
+    rolldown['throttle and brake == 0 or na'] = ((rolldown['throttle'] == 0) & \
+                                            (rolldown['mechanical_brake_state'] == 0)) | \
```
[the remark this answers is not in the document]
Comment (ufshk) [reply dynamics/rolldowndata/rolldown/clean_rolldown_data.py:28]: Pandas uses bitwise operators. Logical operators here resulted in errors.

File: dynamics/rolldowndata/rolldown/clean_rolldown_data.py (hunk 2 above)
Comment (ufshk) [reply dynamics/rolldowndata/rolldown/clean_rolldown_data.py:57]: before this line it had two rows of column headers, I drop one level

**Pull request merged**